raiden/tests/integration/fixtures/raiden_network.py

```diff
@@ -279,3 +280,21 @@ def raiden_network(
     yield raiden_apps
 
     shutdown_apps_and_cleanup_tasks(raiden_apps)
+
+
+class RestartNode:
+    def __init__(self):
+        self.async_result: Optional[AsyncResult] = None
+
+    def link_exception_to(self, result: AsyncResult) -> None:
+        self.async_result = result
+
+    def __call__(self, app: App) -> None:
+        app.start()
+        if self.async_result is not None:
```

Comment:
Should this be linked before the call to `start` in case something goes wrong there?

Reply:
Yes, good idea. I'll change it.